clusterDeployer.py

```diff
@@ -119,15 +119,19 @@ def teardown(self) -> None:
         removed_macs = []
         names = [x.name for x in self._cc.all_vms()]
         ips = [x.ip for x in self._cc.all_vms()]
-        for e in q[-1][0][1:]:
-            if e.attrib["name"] in names or e.attrib["ip"] in ips:
-                mac = e.attrib["mac"]
-                name = e.attrib["name"]
-                ip = e.attrib["ip"]
-                pre = "virsh net-update default delete ip-dhcp-host"
-                cmd = f"{pre} \"<host mac='{mac}' name='{name}' ip='{ip}'/>\" --live --config"
-                logger.info(self._local_host.hostconn.run(cmd))
-                removed_macs.append(mac)
+        ip_tree = q.find('ip')
+        if ip_tree:
+            dhcp = ip_tree.find('dhcp')
+            if dhcp:
+                for e in dhcp:
+                    if e.get('name') in names or e.get('ip') in ips:
+                        mac = e.attrib["mac"]
+                        name = e.attrib["name"]
+                        ip = e.attrib["ip"]
+                        pre = "virsh net-update default delete ip-dhcp-host"
+                        cmd = f"{pre} \"<host mac='{mac}' name='{name}' ip='{ip}'/>\" --live --config"
+                        logger.info(self._local_host.hostconn.run(cmd))
+                        removed_macs.append(mac)
```

Comment:
Does this work when you have two clusters deployed on the same provision host and you are cleaning up one of the clusters? The previous code used to work with two clusters.

Reply:
It should. The only difference compared to before the patch is the way the loop is built.
Instead of using q[-1][0][1:] i.e.  instead of supposing that ip is the last element [-1] of the network tree, and instead of guessing the dhcp elements, we now search for them, still based on net-dumpxml output. 
Final result is that we go through all dhcp elements containing name or ip of our vms, as we were doing before (when net-dumpxml was as expected ...)